### PR TITLE
Add a small README

### LINKED ISSUE #22: Write a proper README
^

### PR DESCRIPTION
**Pull request by vkleen**
Depends on #29. Closes #22.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,71 @@
...
+sense. Finally, there is some housekeeping to do for dispatching between these
+two functionalities.
+
+## Limitations
+
+- JSON schema comes in many dialects. We rely on [schemars][schemars] to
+  interpret schemas and this way should support any draft specification and
+  dialect that [schemars][schemars] does.
+- The `format` keyword of JSON schema is currently ignored, [#24][i24]
+- Error reporting in contracts converted from predicates can be a bit hit or miss, [#27][i27]
```
Comment (yannham) [inline README.md:67]: Should we mention the partial support for completion with the Nickel LSP? And the fact that the generated contract might be eager?

File: README.md [hunk 2]
```diff
@@ -1,3 +1,71 @@
...
+```shell
+cargo build --release
+```
+
+in a checkout of this repository. The resulting binary will be `./target/
+release/json-schema-to-nickel` and can also be run using `cargo run --release`.
+Alternatively, you can build it using `nix` by running
+
+```shell
+nix build github:nickel-lang/json-schema-to-nickel
```
[the remark this answers is not in the document]
Comment (vkleen) [reply README.md:19]: That's a much better idea, thanks 👍

File: README.md [hunk 3]
```diff
@@ -1,3 +1,76 @@
...
+JSON schema. A JSON schema encodes a predicate on JSON objects, that is, a
+function from JSON objects returning `true` or `false`. Such a function can also
+be written in Nickel and a big part of `json-schema-to-nickel` is mechanically
+generating such a predicate using a [Nickel library](./lib/predicates.ncl).
+However, such predicates are not ideal for use in Nickel and many aspects
+of JSON schema can be encoded as Nickel record contracts. The second major
+part of `json-schema-to-nickel` is figuring out when a JSON schema can be
+represented as a record contract, and generating this contract when it makes
+sense. Finally, there is some housekeeping to do for dispatching between these
+two functionalities.
```
Comment (Radvendii) [inline README.md:55]: 1. I would make it clear that this is not currently optimal. There are certainly json schemas that could be turned into record contracts that we don't (yet).
2. I would also say why we want to generate contracts this way. i.e. laziness, and LSP integration. I see you explain it below, but I think it belongs in this section too.

**Review by yannham — APPROVED** [review]
(Approving this to not block this PR, although @vkleen might want to take the last review into account, it's in a totally mergeable state already)

**Pull request merged**